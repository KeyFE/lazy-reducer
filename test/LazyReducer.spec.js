import { LazyReducer, lazyReducerEnhancer } from '../lib'
import { createStore, combineReducers } from 'redux'
import renderer from 'react-test-renderer'
import React, { Component } from 'react'
import { connect, Provider } from 'react-redux'

@connect(state => {
    return {
        store: state
    }
})
class InnerComponent extends Component {
    render() {
        return <div>InnerComponent {JSON.stringify(this.props)}</div>
    }
}

@connect(state => {
    return {
        store: state
    }
})
class DelayContainer extends Component {
    constructor(props) {
        super(props)
        this.state = {}
        setTimeout(() => {
            this.setState({
                show: true
            })
        }, 200)
    }

    render() {
        if (this.state.show) {
            return (
                <LazyReducer
                    reducers={{
                        nameX: function(state = null, action) {
                            return state
                        }
                    }}
                    x = 'xxx'
                >
                    <InnerComponent />
                </LazyReducer>
            )
        } else {
            return <div>DelayContainer {JSON.stringify(this.props)}</div>
        }
    }
}

function waitFor(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay)
    })
}

test('LazyReducer', async () => {
    const syncReducers = {
        a: (state = 'a', action) => {
            return state
        },
        b: (state = 'x', action) => {
            return state
        }
    }
    const store = createStore(combineReducers(syncReducers), {}, lazyReducerEnhancer(syncReducers))
    const App = (
        <Provider store={store}>
            <DelayContainer />
        </Provider>
    )
    const app = renderer.create(App)
    expect(app.toJSON()).toMatchSnapshot()
    await waitFor(300)
    expect(app.toJSON()).toMatchSnapshot()
})
