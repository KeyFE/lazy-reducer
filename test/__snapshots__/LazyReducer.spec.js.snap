// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyReducer 1`] = `
<div>
  DelayContainer 
  {"store":{"a":"a","b":"x"}}
</div>
`;

exports[`LazyReducer 2`] = `
<div>
  InnerComponent 
  {"x":"xxx","store":{"a":"a","b":"x","nameX":null}}
</div>
`;
